app/models/ability.rb

```diff
@@ -93,7 +93,7 @@ def initialize(user)
   end
 
   def signed_in?(user)
-    user.persisted?
+    user.persisted? && user.confirmed?
```

Comment:
I'm not a huge fan of devise (because it does so many magic things under the hood). I feel like an unpersisted user should never be confirmed. But since we inject the user object, we cannot guarantee that unless we overwrite `User#confirmed?` to `super && persisted?` Is it worth it? :thinking:  (sorry, just thinking aloud)

Reply:
Oh, hm, no idea. Good point. I should look into what `#persisted?` does...

Reply:
A bit late, but I'm not sure if I understand your comment @carpodaster? What does the `Abilities` file have to do with Devise's magic?  
To me this looks more like an accidental mixing in things that Devise already does into role management... I for instance don't quite understand why `signed_in` is even a question within authorization - it's a Devise concern - and setting it equal to `persisted` is also wrong (persisted user can also be signed out...)

Maybe this thing just has the wrong name and we should simply use `user.persisted?` everywhere within the abilities where it makes sense? What do you think @carpodaster & @ramonh?